### ISSUE TITLE
Whitespace and special characters in env var values can still cause restart loops

### ISSUE THREAD

**Issue by CameronDiver** [balena-os/balena-supervisor#965]
Despite the handling code for this, there is still instances of environment variable values causing the supervisor to think that the config has changed and restart the container repeatedly. The latest instance was a variable like so:
```
VAR="\t value"
```

**Comment by dekimsey** [comment]
I was able to trigger this issue by pasting a multiline value in the web API. Though it appeared on one line in the web interface, the newlines were encoded and broke the resin-supervisor.service, triggering a restart loop.

```
Apr 08 00:06:29 fc5dac6 systemd[1]: resin-supervisor.service: Failed to load environment files: Invalid argument
Apr 08 00:06:29 fc5dac6 systemd[1]: resin-supervisor.service: Failed to run 'start-pre' task: Invalid argument
Apr 08 00:06:29 fc5dac6 systemd[1]: resin-supervisor.service: Failed with result 'resources'.
Apr 08 00:06:29 fc5dac6 systemd[1]: Failed to start Balena supervisor.
```

**Comment by CameronDiver** [comment]
@dekimsey this looks to me like a separate issue completely, and I'd hazard a guess that your SD card is reaching the end of its life, causing corruption (assuming an rpi3 here).

To check this you can run `md5sum --quiet -c /resinos.fingerprint` from the host OS, and there should be a maximum of ~4 failures. If this isn't the case, please make a thread on the forums and we'll be happy to help you there.

For this issue, if you could provide me with the steps used to apply this environment variable, and it's value so I can easily reproduce that would be extremely helpful :)

**Comment by dekimsey** [comment]
@CameronDiver Interesting! That file looked so bizarre I was thinking it was some sort of sed transformations or binary encoding that were screwed up as I had very recently copy/pasted a multiline env variable into the UI.

As to whether I triggered this issue, my workflow was as such:
1. Generate SSH private key
2. Copy from terminal into web interface as a Device Service Variable
(Note the multline string appears to be a one-line)
4. Submit

I think it unlikely I somehow triggered this behavior. But I was looking up restart loops and this was the only recent change I had performed.

Given your feedback, I doubt I triggered this actual issue. But since the md5sum test passed, something clearly went bad on my system.  I only received two errors, so it looks like I will be opening a thread on the issue.

Thank you
